cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/template/etl/batch/ETLTPFSTest.java

```diff
@@ -0,0 +1,141 @@
+
+    GenericRecord record = new GenericRecordBuilder(avroSchema)
+      .set("int", Integer.MAX_VALUE)
+      .build();
+
+    String filesetName = "tpfs";
+    DatasetAdmin datasetAdmin = addDatasetInstance("timePartitionedFileSet", filesetName, FileSetProperties.builder()
+      .setInputFormat(AvroKeyInputFormat.class)
+      .setOutputFormat(AvroKeyOutputFormat.class)
+      .setInputProperty("schema", avroSchema.toString())
+      .setOutputProperty("schema", avroSchema.toString())
+      .setEnableExploreOnCreate(true)
+      .setSerDe("org.apache.hadoop.hive.serde2.avro.AvroSerDe")
+      .setExploreInputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat")
+      .setExploreOutputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat")
+      .setTableProperty("avro.schema.literal", (avroSchema.toString()))
+      .build());
+    DataSetManager<TimePartitionedFileSet> fileSetManager = getDataset(filesetName);
+    TimePartitionedFileSet tpfs = fileSetManager.get();
+
+    TransactionManager txService = getTxService();
+    Transaction tx1 = txService.startShort(100);
+    TransactionAware txTpfs = (TransactionAware) tpfs;
+    txTpfs.startTx(tx1);
+
+    fileSetManager.get().addPartition(System.currentTimeMillis(), "directory", ImmutableMap.of("key1", "value1"));
+    Location location = fileSetManager.get().getPartitionByTime(System.currentTimeMillis()).getLocation();
+    location = location.append("file.avro");
+    FSDataOutputStream outputStream = new FSDataOutputStream(location.getOutputStream(), null);
+    DataFileWriter dataFileWriter = new DataFileWriter<>(new GenericDatumWriter<GenericRecord>(avroSchema));
+    dataFileWriter.create(avroSchema, outputStream);
+    dataFileWriter.append(record);
+    dataFileWriter.flush();
+
+    txTpfs.commitTx();
+    txService.canCommit(tx1, txTpfs.getTxChanges());
+    txService.commit(tx1);
+    txTpfs.postTxCommit();
+
+    String newFilesetName = filesetName + "_op";
+    ETLBatchConfig etlBatchConfig = constructTPFSETLConfig(filesetName, newFilesetName);
+
+    AdapterConfig newAdapterConfig = new AdapterConfig("description", TEMPLATE_ID.getId(),
+                                                       GSON.toJsonTree(etlBatchConfig));
+    Id.Adapter newAdapterId = Id.Adapter.from(NAMESPACE, "sconversion1");
+    AdapterManager tpfsAdapterManager = createAdapter(newAdapterId, newAdapterConfig);
+
+    tpfsAdapterManager.start();
+    tpfsAdapterManager.waitForOneRunToFinish(4, TimeUnit.MINUTES);
+    tpfsAdapterManager.stop();
+
+    DataSetManager<TimePartitionedFileSet> newFileSetManager = getDataset(newFilesetName);
+    TimePartitionedFileSet newFileSet = newFileSetManager.get();
+
+    List<GenericRecord> newRecords = readOutput(newFileSet, EVENT_SCHEMA_WITHOUT_HEADER);
+    Assert.assertEquals(1, newRecords.size());
+  }
+
+  private org.apache.avro.Schema convertSchema(Schema cdapSchema) {
```

Comment:
we need to  convert it because the eventSchemaWithoutHeader (which in this commit has been changed to eventSchema) takes in a Schema of StructuredRecords